src/protocols/secure_channel/CASESession.cpp

```diff
@@ -224,12 +233,29 @@ class CASESession::WorkHelper
         bool cancel = false;
         // Execute callback in background thread; data must be OK with this
         helper->mStatus = helper->mWorkCallback(helper->mData, cancel);
+
+        // Check the status and call OnWorkFailedCallback
+        if (helper->mStatus != CHIP_NO_ERROR)
+        {
+            if (auto * session = helper->mSession.load())
+            {
+                (session->*(helper->mOnWorkFailedCallback))(helper->mStatus);
```

Comment:
Please introduce helper variables to improve legibility of this incantation if possible